matrix/matrix.py

```diff
@@ -0,0 +1,176 @@
+from typing import Union
+
+
+class Matrix:
+
+    def __init__(self, matrix: tuple):
+        Matrix.is_valid_matrix(matrix)
+        self.matrix = matrix
+
+    @property
+    def tuples(self):
+        return self.matrix
+
+    def __str__(self):
+        return str(self.matrix)
+
+    def __repr__(self):
+        return f"Matrix({self.matrix})"
+
+    def __len__(self):
+        return len(self.matrix)
+
+    def __getitem__(self, i):
+        return self.matrix[i]
+
+    def __hash__(self):
+        return hash(row for row in self)
+
+    def __neg__(self):
+        return Matrix(tuple(tuple(-item for item in row) for row in self))
+
+    def __eq__(self, other):
+        if type(other) is not Matrix:
+            raise TypeError("Argument is not of Matrix type.")
+
+        if len(self) != len(other):
+            return False
+
+        if self is other:
+            return True
+
+        if any(s_row != o_row for s_row, o_row in zip(self, other)):
+            return False
+
+        return True
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __add__(self, other):
+        if type(other) is not Matrix:
+            raise TypeError("Argument is not of Matrix type.")
+
+        if len(self) != len(other):
+            raise ValueError("Matrix must be of the same order.")
+
+        if self is other:
+            return self * 2
+
+        return Matrix(tuple(tuple((sum(cell_pair) for cell_pair in zip(s_row, o_row)))
+                            for s_row, o_row in zip(self, other)))
```

Comment:
Why making your variables names more shorter here (there are more several places)? What it gives you? 
Consider:
`return self(tuple(tuple((sum(cell_pair) for cell_pair in zip(self_row, other_row)))
                            for self_row, other_row in zip(self, other)))`